### ISSUE TITLE
Line numbers and function names don't match

### ISSUE THREAD

**Issue by shashank-srikant** [ConsenSys/mythril-classic#424]
## Description

Running `mythril -xo json <filename>` produces a report in which sometimes the function names and the corresponding line numbers just don't make sense.
I understand that this is information that's sourced from the solidity binary, etc. But just wanted to make sure this is expected behavior.

## Steps to reproduce

For instance, in  https://etherscan.io/address/0x1c3bed5bd4b0e6a8389efd0d2876f948a907ab5b#code, for the contract `MMR`, one of the errors Mythril reports is - 
```
[0]
address : 573
code : "his.balance); return; } MiniMeToken toke"
debug : ""
description : "This contract executes a message call to to another contract. Make sure that the called contract is trusted and does not execute user-supplied code."
filename : "<filename>"
function : "fallback"
lineno : 507
title : "Message call to external contract"
type : "Informational"
```
Not sure which function `fallback` refers to in the entire contract/ source file.
The line number `507` corresponds to the function `claimToken`.
Can this mismatch please be explained.

**Comment by norhh** [comment]
Hi shashank, Thanks for pointing it out, Here `fallback` refers to the solidity [fallback function](https://www.bitdegree.org/learn/solidity-fallback-functions/). After looking at the contract it doesn't look like the function `claimToken` is a `fallback` function.

**Comment by norhh** [comment]
can you share them, Thanks for the help

**Comment by shashank-srikant** [comment]
Here are a few.
Format
```
Line1: solidity_source_addr:contract_name
Line2: Name of function reported by Mythril
Line3: Line number reported by Mythril
Line4: Name of function the line number actually corresponds to
--
```
i see how `add` and `safeAdd` get flagged, and there's a line number/function name mismatch. but a few of them aren't clear why.

```
--
0xed5a90efa30637606ddaf4f4b3d42bb49d79bd4e:MultiSigWalletWithDailyLimit
removeOwner(address)
378
CLASS_MEMBER
--
0x2e5e995bdaa495abaf2af499d682b0277d4dd66b:MintedEthCappedCrowdsale
isPricingSane()
1100
isCrowdsale
--
0x2e5e995bdaa495abaf2af499d682b0277d4dd66b:MintedEthCappedCrowdsale
isPricingSane()
1100
isCrowdsale
--
0x416993d2384d9b82687f34f7fea29f6fb2c6c56d:Market
fallback
1222
computeEtherAmount
--
0xd907Bd07813Ba91A9143E7c98340E53755d575A5:UnilotTailEther
revoke()
349
UnilotTailEther
--
0xfb1fb0eafdbaa51f8ba1332e43599e75b7cc8e2c:BurnableToken
transfer(address,uint256)
51
add
--
0x75096895714C15a0E31686b73eFD16FB108F4aDE:EPXCrowdsale
fallback
68
safeAssert
--
0x75096895714C15a0E31686b73eFD16FB108F4aDE:EPXCrowdsale
fallback
161
checkPrice
--
0x2e5e995bdaa495abaf2af499d682b0277d4dd66b:MintedEthCappedCrowdsale
finalize()
1124
CLASS_MEMBER
--
0x2e5e995bdaa495abaf2af499d682b0277d4dd66b:MintedEthCappedCrowdsale
getState()
1125
CLASS_MEMBER
--
0x2e5e995bdaa495abaf2af499d682b0277d4dd66b:MintedEthCappedCrowdsale
preallocate(address,uint256,uint256)
859
investWithSignedAddress
--
0x2e5e995bdaa495abaf2af499d682b0277d4dd66b:MintedEthCappedCrowdsale
preallocate(address,uint256,uint256)
864
investWithSignedAddress
--
0x2e5e995bdaa495abaf2af499d682b0277d4dd66b:MintedEthCappedCrowdsale
preallocate(address,uint256,uint256)
17
safeAdd
--
0x2e5e995bdaa495abaf2af499d682b0277d4dd66b:MintedEthCappedCrowdsale
preallocate(address,uint256,uint256)
18
safeAdd
--
0x9aed7a25f2d928225e6fb2388055c7363ad6727b:DSAuth
setOwner(address)
135
setAuthority
--
0xb5dbd316fa0368a99b3ca19597a5b22c70349b76:StandardToken
increaseApproval(address,uint256)
275
CLASS_MEMBER
--
0xb5dbd316fa0368a99b3ca19597a5b22c70349b76:StandardToken
increaseApproval(address,uint256)
275
CLASS_MEMBER
--
0xa7a3de21f9505eaa8a4d34ae2d5497f8d5df35c2:EXGROUP
burnFrom(address,uint256)
172
CLASS_MEMBER
--
0x9ef15d682f48b23bed7612bbc21b61160e628ea5:StandardToken
increaseApproval(address,uint256)
94
add
--
0x9ef15d682f48b23bed7612bbc21b61160e628ea5:StandardToken
increaseApproval(address,uint256)
94
add
--
0x8aa3aeb44af42d4ff0f68066cace6658a076dee8:StandardToken
increaseApproval(address,uint256)
46
add
--
```

**Comment by shashank-srikant** [comment]
also, does Consensys have a bug-bounty program on this product, for me to actually benefit from combing through these nuances/issues :)

**Comment by b-mueller** [comment]
@shashank-srikant no bug bounty program yet, but we're currently working with Gitcoin to set something up. We'll keep you posted!

**Comment by norhh** [comment]
In nearly all cases it is outputting the root function in the function path, But it is preferable to recheck this again after #471

**Comment by norhh** [comment]
should be fixed by #949 , if the bug still persists, please feel free to reopen this.
